crates/sui-core/src/transaction_orchestrator.rs

```diff
@@ -236,30 +236,52 @@ where
                         false,
                     ))));
                 }
-                let executable_tx = VerifiedExecutableTransaction::new_from_quorum_execution(
-                    transaction,
-                    effects_cert.executed_epoch(),
-                );
 
-                match Self::execute_finalized_tx_locally_with_timeout(
-                    &self.validator_state,
-                    &executable_tx,
-                    &effects_cert,
-                    objects,
-                    &self.metrics,
-                )
-                .await
-                {
-                    Ok(_) => Ok(ExecuteTransactionResponse::EffectsCert(Box::new((
+                // TODO: local execution for shared-object txns is disabled due to the fact that
+                // it can cause forks on transactions that read child objects from read-only
+                // parent shared objects.
+                //
+                // This can be re-enabled after MVCC for child objects is merged.
```

Comment:
Maybe we can have a debug / info log entry that the transaction is executed via local execution path? It will make confirming related issues easier in future. Or I can do that in a separate PR.